Directory.Build.targets

```diff
@@ -67,14 +67,24 @@
     <PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(IsMgmtLibrary)' == 'true'">
+  <ItemGroup Condition="'$(IsMgmtLibrary)' == 'true' and '$(IsTestGenerationSrcProject)' == 'true'">
     <PackageReference Include="Azure.Core" />
     <PackageReference Include="System.Text.Json" />
     <PackageReference Include="System.ClientModel" />
     <PackageReference Include="Azure.ResourceManager" />
 
     <!-- TODO: Review these file references-->
     <Compile Include="$(AzureCoreSharedSources)AzureResourceProviderNamespaceAttribute.cs" LinkBase="Shared" />
+    <!-- Include management shared code-->
+    <Compile Include="$(MSBuildThisFileDirectory)/src/assets/Management.Shared/**/*.cs" Link="Shared/Management/%(RecursiveDir)/%(Filename)%(Extension)" />
+  </ItemGroup>
+
+  <ItemGroup Condition ="'$(IsMgmtLibrary)' == 'true' and '$(IsTestGenerationTestProject)' == 'true'">
```

Comment:
Looks like for non-test projects of mgmt library, we don't need to add packages?

Reply:
hello @archerzz above this, we will add dependent packages for src-project of mgmt SDKs.